java/query/src/test/java/sleeper/query/recordretrieval/InMemoryLeafPartitionRecordRetriever.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2022-2023 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.query.recordretrieval;
+
+import sleeper.core.iterator.CloseableIterator;
+import sleeper.core.record.Record;
+import sleeper.core.schema.Schema;
+import sleeper.query.model.LeafPartitionQuery;
+
+public class InMemoryLeafPartitionRecordRetriever implements LeafPartitionRecordRetriever {
+
+    @Override
+    public CloseableIterator<Record> getRecords(Schema dataReadSchema, Schema tableSchema,
+            LeafPartitionQuery leafPartitionQuery) throws RecordRetrievalException {
+        // TODO Auto-generated method stub
+        throw new UnsupportedOperationException("Unimplemented method 'getRecords'");
+    }
```

Comment:
If this was implemented with a field `Map<String, List<Record>>` where the key is the filename matching the FileInfo from the state store, it should be possible to return the relevant records here and have a method to add a file with a given set of records. This could be used to test the behaviour of QueryExecutor.